src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -160,6 +161,59 @@ public ResponseDTO fulfilmentRequestBySMS(SMSFulfilmentRequestDTO requestBodyDTO
     return response;
   }
 
+  @Override
+  public CaseDTO createCaseForNewAddress(NewCaseRequestDTO caseRequestDTO) throws CTPException {
+    CaseType caseType = caseRequestDTO.getCaseType();
+
+    // Validate that case type and address match
+    String censusAddressType;
+    if (caseRequestDTO.getEstabType() == EstabType.OTHER) {
+      // Can't get an address type from the estab so it'll need to be the same as the case type
+      censusAddressType = caseType.name();
+    } else {
+      AddressType addressType = caseRequestDTO.getEstabType().getAddressType().get();
+      if (!addressType.name().equals(caseType.name())) {
+        throw new CTPException(
+            Fault.BAD_REQUEST,
+            "Mismatch between caseType and the address type of the nameed estabType. "
+                + "Expected caseType of '"
+                + addressType.name()
+                + "'");
+      }
+      censusAddressType = addressType.name();
+    }
+
+    // Reject if CE with non-positive number of residents
+    if (caseRequestDTO.getCaseType() == CaseType.CE && caseRequestDTO.getCeUsualResidents() <= 0) {
```

Comment:
Not all combinations covered here in the unit tests - consider enhancing for better coverage.

Reply:
Added testcase to cover CE with positive number of residents